src/backend/cpu/Param.hpp

```diff
@@ -0,0 +1,100 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+#include <af/defines.h>
+#include <af/dim4.hpp>
+#include <backend.hpp>
+
+namespace cpu
+{
+
+using af::dim4;
+
+template<typename T>
+class CParam
+{
+private:
+    const T *ptr;
+
+public:
+    dim4 dims;
+    dim4 strides;
```

Comment:
I would like these to be private or at the very least const. There is no reason for the dims and the strides to change once the [C]Param object is created.